packages/wallet/tests/wallet.spec.ts

```diff
@@ -317,13 +317,55 @@ describe('Wallet integration', function () {
 
             expect(tx.data).to.contain("00fffff")
           })
+
           it('Should estimate gas for transaction with 0 gasLimit and revertOnError false', async () => {
             await new ethers.ContractFactory(
               MainModuleArtifact.abi,
               MainModuleArtifact.bytecode,
               signer
             ).deploy(wallet.address)
           })
+
+          it('Should be able to update the config for a wallet with many signers', async () => {
+            // first, we try just one signer
+
+            const signers = wallet.config.signers
+            while (signers.length < 1) {
+              signers.push({
+                address: ethers.Wallet.createRandom().address,
+                weight: 1
+              })
+            }
+
+            const newConfig = {
+              threshold: 1,
+              signers
+            }
+
+            let tx = (await wallet.updateConfig(newConfig, undefined, true))[1]
+            let receipt = await tx.wait()
+            expect(receipt.status).to.equal(1)
+            const gasLimit1 = tx.gasLimit
+
+            // next, we try 100 signers
+
+            while (signers.length < 100) {
```

Comment:
Can we test 1000 signers and remove the 2x factor when calculating the gas limit above? I'd rather have a tighter gas limit estimate.

Reply:
or use 1.25x factor if we feel we need the safety? as well, if we fail-fast, and catch errors, then we'll know at least when we hit issues with gas. In the future, we should always remember when something breaks to always just check the relayer wallet